include/Piece.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <iostream>
+#include <vector>
+using std::vector;
+#include <string>
+#include "Move.h"
+#include "Location.h"
+
+
+// Forward declaration of Board
+class Board;
+
+enum Color { WHITE, BLACK ,NONE};
+
+class Piece {
+protected:
+    Color color;
+    Location location;
+    char symbol = 'x';
+    Board* board;
```

Comment:
saving a pointer for the borad as private members.
bad way for checking collisions, pieces have full access for the board